transcriptic/config.py

```diff
@@ -248,14 +243,20 @@ def token(self):
             raise ValueError("token is not set.")
 
     @token.setter
-    def token(self, value):
+    def token(self, value: str):
         if self.cookie is not None:
             warnings.warn(
                 "Cookie and token authentication is mutually "
                 "exclusive. Clearing cookie from headers"
             )
             self.update_headers(**{"Cookie": None})
-        self.update_headers(**{"X-User-Token": value})
+
+        if value is not None:
+            is_bearer_token = value.startswith("Bearer")
```

Comment:
hmm.. yea.. This pattern isn't ideal and is not very extensible if/when we need more auth methods. It also starts getting brittle if we want to do further validations on the type/schema of token passed through.

This doesn't have to be done now, but really what I think we should be doing is to move our server towards a more traditional/official Auth method (e.g. using Basic auth for passing in tokens) instead of a custom `X-User-Token` field.

Correspondingly on the client side, we can then use the `requests.get(..., auth=SomeAuth(...)` pattern and a more general `auth` field. 
In this case, we would construct a `Bearer(requests.auth.AuthBase)` class and pass it into the `requests.get(..., auth=Bearer(...))` field.

Reply:
Yes, agreed 100%. Basic auth is similar to what we're doing with `x-user-token`. With Basic Auth, I believe the client would send the username & password as a base64 encoded string. We should probably use OAuth-based Authorization code grant type instead of Basic authentication for user authentication/authorization and get rid of basic authentication altogether.

Reply:
Let's add a quick sanity check here on the Bearer token value here. I'm not sure if we're following JWT or any specific syntax, but at the very least, it should match `https://tools.ietf.org/html/rfc6750`, i.e. contiguous alphanumeric with "-" / "." / "_" / "~" / "+" / "/" characters ending 0 or more `=`

Adding a test would be nice